cla_public/apps/checker/fields.py

```diff
@@ -161,10 +161,22 @@ def __init__(self, label=None, validators=None, min_val=0, max_val=9999999999, *
         self.min_val = min_val
         self.max_val = max_val
 
+    def convert_amount_to_utf8(self, amount):
+        # Covert pounds side of the input into UTF-8 format
+        try:
+            amount = amount.decode("utf-8")
+        except UnicodeEncodeError:
+            # Input is already in UTF-8 format
+            pass
+
+        return amount
+
     def process_formdata(self, valuelist):
         if valuelist:
             pounds, _, pence = valuelist[0].strip().partition(".")
```

Comment:
I think getting pounds and pence from valuelist would work nicely as its own function. All the stripping, partitioning, decoding and regex subbing in one place, and then that function has its own meaningful purpose (rather than just a try/catch around an existing function) which could help a reader understand the code flow.